Sources/XcodeGen/main.swift

```diff
@@ -44,8 +44,18 @@ func generate(spec: String, project: String, isQuiet: Bool, justVersion: Bool) {
         let project = try projectGenerator.generateProject()
 
         logger.info("⚙️  Writing project...")
+        
         let projectFile = projectPath + "\(spec.name).xcodeproj"
-        try project.write(path: projectFile, override: true)
+        let tempPath = Path.temporary + "XcodeGen_\(Int(NSTimeIntervalSince1970))"
+        try? tempPath.delete()
+        if projectFile.exists {
+            try projectFile.copy(tempPath)
+        }
+        try project.write(path: tempPath, override: true)
+        try? projectFile.delete()
+        try tempPath.copy(projectFile)
+        try? tempPath.delete()
```

Comment:
We should trap signals to ensure deletion, but it's under tmp dir anyways, so I guess we can leave it as future TODO.